### PR TITLE
feat: TS Config optimisation settings

### PR DESCRIPTION
**Pull request by ariskemper**

### CODE REVIEW

**Comment by fabian-hiller** [comment]
Thank you! I plan to review and merge this PR in the next days.

**Comment by ariskemper** [comment]
@fabian-hiller you could check difference in parse time and check time with `tsc --extendedDiagnostics`

File: library/tsconfig.json [hunk 1]
```diff
@@ -6,7 +6,15 @@
...
     "noEmit": true
   },
-  "include": ["."]
+  "include": [
+    "src/**/*.ts",
+    "mod.ts",
+    "tsup.config.ts",
+    "vite.config.ts"
+  ],
+  "exclude": ["node_modules"]
```
Comment (ariskemper) [inline library/tsconfig.json:19]: @fabian-hiller will remove this, because it is done by default.

File: library/tsconfig.json [hunk 2]
```diff
@@ -6,7 +6,14 @@
     "module": "ESNext",
     "moduleResolution": "node",
     "allowImportingTsExtensions": true,
+    "forceConsistentCasingInFileNames": true,
```
Comment (fabian-hiller) [inline library/tsconfig.json:9]: I think that this is not necessary because `true` is the default value.

**Pull request merged**